src/functions.php

```diff
@@ -69,3 +71,41 @@ function iterable_key_first($iterable)
 
     return null;
 }
+
+/**
+ * Yield sets of items from a sorted iterator grouped by a specific criterion gathered from a callback
+ *
+ * The iterator must be sorted by the criterion. The callback must return at least the criterion,
+ * but can also return value and key in addition.
+ *
+ * @param Iterator $iterator
+ * @param callable(mixed $value, mixed $key): array{0: mixed, 1?: mixed, 2?: mixed} $groupBy
+ *
+ * @return Generator
+ */
+function yield_groups(Iterator $iterator, callable $groupBy): Generator
+{
+    if (! $iterator->valid()) {
```

Comment:
This requirement (the passed `$iterator` being an instance of `Iterator`) does not allow to easily pass objects that implement the `IteratorAggregate` interface, as our IPL orm class `Query` does. You now have to explicitly call the `getIterator()` method. Can't we make this a little less strict?

Reply:
Sure, `Traversable` + `IteratorIterator`.